### PR TITLE
feat: add allocator support

### PR DESCRIPTION
**Pull request by ncitron**

### CODE REVIEW

File: jolt-sdk/src/lib.rs [hunk 1]
```diff
@@ -66,3 +68,54 @@ impl Proof {
...
+
+    pub fn free_memory(&self) -> usize {
+        heap_start() + (self.offset.get() as usize)
+    }
+
+    pub fn fake_alloc(&self, layout: Layout) -> *mut u8 {
+        unsafe {
+            self.alloc(layout)
+        }
+    }
```
Comment (moodlezoup) [inline jolt-sdk/src/lib.rs:102]: what is this for?

File: jolt-sdk/src/lib.rs [hunk 2]
```diff
@@ -66,3 +68,54 @@ impl Proof {
...
+        alloc_start as *mut u8
+    }
+
+    unsafe fn dealloc(&self, _ptr: *mut u8, _layout: Layout) {
+        
+    }
+}
+
+fn align_up(addr: usize, align: usize) -> usize {
+    (addr + align - 1) & !(align - 1)
```
Comment (moodlezoup) [inline jolt-sdk/src/lib.rs:120]: should we be asserting that `align` is a power of two here?

Comment (ncitron) [reply jolt-sdk/src/lib.rs:120]: Align is guaranteed to be a power of two so no need.

File: jolt-sdk/src/lib.rs (hunk 1 above)
Comment (ncitron) [reply jolt-sdk/src/lib.rs]: Oh my bad that's from something I was testing. Removed.

**Review by moodlezoup — APPROVED** [review]

**Pull request merged**